### ISSUE TITLE
[BM2SM request] Forcibly save sm files using UTF-8 encoding format

### ISSUE THREAD

**Issue by sillybear** [SaxxonPike/scharfrichter#7]

**Comment by SaxxonPike** [comment]
@sillybear  I like this suggestion. Stepmania supports reading these, yes?

**Comment by sillybear** [comment]
Yeah any version. Default BM2SM encoding uses ANSI right? I have to manually resave files using utf-8 via notepad to make those non-Roman and special characters display correctly.

**Comment by SaxxonPike** [comment]
According to StepmaniaSM.cs - the default encoding from the .NET `StreamWriter` is used. Which should already be UTF-8 according to this MSDN doc: https://docs.microsoft.com/en-us/dotnet/api/system.io.streamwriter?view=netframework-4.7.2

However! This does not write the byte order mark. Reading further in the MSDN documentation, we see this:
_This behavior is different from the behavior provided by the encoding object in the Encoding.UTF8 property. To specify a BOM and determine whether an exception is thrown on invalid bytes, use a constructor that accepts an encoding object as a parameter, such as StreamWriter(String, Boolean, Encoding) or StreamWriter._

So, when we use `StreamWriter`, we should specify `Encoding.UTF8` in the constructor. This looks like a simple change.
